spatialos/UnrealGDK
# [UNR-2563][MS] Adding spatial auth into the deployment process.

**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
If you are have not gone through the spatial auth process before trying to deploy, the deployment will fail. Adding the spatial auth step to the the deployment process. Also added to the failed deployment toast informing the user that additional detail regarding the failure can be seen in the logs. This is not a perfect solution but is a quick improvement for now.  

How can this be verified by QA?
1. Open UnrealExampleProject.
2. Open CMD and navigate to the spatial folder inside the UnrealExampleProject.
3. Enter into the CMD: "spatial auth logout".
4. In the UE4 Editor, click the deploy button and continue through the deployment flow.

Result: The deployment should not fail due to not beening authed into spatial. You should see a tab open in your default browser stating that the auth processing has been initiated. 

#### Documentation
Updated changelog.

#### Primary reviewers
@joshuahuburn @aleximprobable
